src/Identity/Core/src/IdentityServiceCollectionExtensions.cs

```diff
@@ -109,6 +114,47 @@ public static class IdentityServiceCollectionExtensions
         return new IdentityBuilder(typeof(TUser), typeof(TRole), services);
     }
 
+    /// <summary>
+    /// Adds a set of common identity services to the application to support <see cref="IdentityApiEndpointRouteBuilderExtensions.MapIdentityApi{TUser}(IEndpointRouteBuilder)"/>
+    /// and configures authentication to support identity bearer tokens and cookies.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/>.</param>
+    /// <returns>The <see cref="IdentityBuilder"/>.</returns>
+    public static IdentityBuilder AddIdentityApiEndpoints<TUser>(this IServiceCollection services)
+        where TUser : class, new()
+        => services.AddIdentityApiEndpoints<TUser>(_ => { });
+
+    /// <summary>
+    /// Adds a set of common identity services to the application to support <see cref="IdentityApiEndpointRouteBuilderExtensions.MapIdentityApi{TUser}(IEndpointRouteBuilder)"/>
+    /// and configures authentication to support identity bearer tokens and cookies.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/>.</param>
+    /// <param name="configure">Configures the <see cref="IdentityOptions"/>.</param>
+    /// <returns>The <see cref="IdentityBuilder"/>.</returns>
+    public static IdentityBuilder AddIdentityApiEndpoints<TUser>(this IServiceCollection services, Action<IdentityOptions> configure)
```

Comment:
Is there any reason we don't add this to the IdentityBuilder API? I was looking at my todo app and I want to call it here:

https://github.com/davidfowl/TodoApi/blob/213325a86a615ac516124896a3c7d3138b665be1/TodoApi/Program.cs#L15-L16

Not on the top level. I dont' want cookies involved.

Reply:
`AddApiEndpoints` is the IdentityBuilder extension method which is called by `AddIdentityApiEndpoints<TUser>`. Everything `AddIdentityApiEndpoints<TUser>` calls is public, so you can just skip the cookie parts and do everything else. That's exactly what I do in [AddIdentityEndpointsBearerOnly in the MapIdenityTests](https://github.com/dotnet/aspnetcore/pull/48595/files#diff-d88f54b4508524c717fc2d147db4e69bf6da267e6f1d8b1cbd3c627a8b88a3acR382-R390 ).

Reply:
```C#
services
            .AddIdentityCore<ApplicationUser>()
            .AddEntityFrameworkStores<ApplicationDbContext>()
            .AddApiEndpoints();
        services
            .AddAuthentication(IdentityConstants.BearerScheme)
            .AddIdentityBearerToken<ApplicationUser>();
```

Why doesn't AddApiEndpoints add the authentication scheme as well?

Reply:
Is there any reason I need to specify the scheme directly? We default to the single scheme if there's only one right?

Reply:
You can use the endpoints with just cookies or just bearer tokens, and I don't think it really makes sense to pick a preferred authentication scheme. So you either call `AddIdentityApiEndpoints<TUser>` to add both schemes, or you add whichever scheme you want manually.

I didn't know about auth picking the default to the single scheme if there's only one. I just tried it, and that is indeed the case.